src/irmin/tree.ml

```diff
@@ -931,9 +931,12 @@ module Make (P : S.PRIVATE) = struct
       match map t with
       | Some m -> Lwt.return (StepMap.is_empty m)
       | None -> (
-          to_value t >|= function
-          | None -> false
-          | Some n -> P.Node.Val.is_empty n )
+          match t.v with
+          | Value (_, _, Some _) -> Lwt.return false
```

Comment:
can avoid a minor allocation here with `Lwt.return_false` to use the precomputed value

Reply:
Indeed. In this and many other places.

https://github.com/mirage/irmin/pull/842